sign/signup/index.js

```diff
@@ -1,100 +1,155 @@
-import { isValidEmail, isValidPwd } from '../../utiles/validation.js';
-
-const emailInput = document.querySelector('#email');
-const pwdInput = document.querySelector('#password');
-const pwdConfirmInput = document.querySelector('#password-confirm');
-const loginBtn = document.querySelector('.btn-login');
-const pwdToggleIcon = document.querySelector('#toggle-eye');
-
-
-const checkEmailValidation = (e) => {
-  const inputField = document.querySelector('.email-field');
+import { isEmail, isPassword } from '/utils/validation.js';
+import { addErrorMessage, removeErrorMessage } from '/utils/error.js';
+import { toggleEyeIcon } from '/utils/toggle-icon.js';
+import { 
+  EMAIL_EMPTY, 
+  EMAIL_INVALID, 
+  EMAIL_USED,
+  PASSWORD_INVALID,
+  PASSWORD_NOT_MATCH
+} from '/constants/error-message.js';
+import {
+  EMAIL_INPUT,
+  EMAIL_ERROR,
+  PASSWORD_INPUT,
+  PASSWORD_ERROR,
+  PASSWORD_CONFIRM_INPUT,
+  PASSWORD_CONFIRM_ERROR,
+  EYE_ICON,
+  EYE_ICON_RE,
+  LOGIN_BUTTON
+} from '/constants/selector.js';
+
+// Email
+const emailInput = document.querySelector(EMAIL_INPUT);
+const emailError = document.querySelector(EMAIL_ERROR);
+
+// Password
+const passwordInput = document.querySelector(PASSWORD_INPUT);
+const passwordError = document.querySelector(PASSWORD_ERROR);
+const pwdConfirmInput = document.querySelector(PASSWORD_CONFIRM_INPUT);
+const pwdConfirmError = document.querySelector(PASSWORD_CONFIRM_ERROR);
+
+// Toggle
+const eyeIcon = document.querySelector(EYE_ICON);
+const eyeIcon_Re = document.querySelector(EYE_ICON_RE);
+
+// LoginButton
+const loginBtn = document.querySelector(LOGIN_BUTTON);
+
+
+// 이메일 유효성 검사 
+function checkEmailValidation() {
   const email = emailInput.value.trim();
-  const errorMessage = inputField.querySelector('.error');
 
   if (email === '') {
-    errorMessage.textContent = '이메일을 입력해주세요.';
-    inputField.classList.add('error');
-  } else if (!isValidEmail(email)) {
-    errorMessage.textContent = '올바른 이메일 주소가 아닙니다.';
-    inputField.classList.add('error');
-  } else if (email === 'test@codeit.com') {
-    errorMessage.textContent = '이미 사용 중인 이메일입니다.';
-    inputField.classList.add('error');
+    addErrorMessage(emailInput, emailError, EMAIL_EMPTY);
+    return false;
+  } else if (!isEmail(email)) {
+    addErrorMessage(emailInput, emailError, EMAIL_INVALID);
+    return false;
+  } else if (isDuplicatedEmail(email)) {
```

Comment:
```suggestion
  } else if (await isDuplicatedEmail(email)) {
```
async 함수라서 await를 앞에 적어주셔야 정확한 boolean이 반환됩니다!